sql/src/main/java/io/crate/analyze/ConstraintsValidator.java

```diff
@@ -22,21 +22,50 @@
 
 package io.crate.analyze;
 
+import io.crate.core.collections.Maps;
 import io.crate.metadata.ColumnIdent;
 import io.crate.metadata.Reference;
 import io.crate.metadata.doc.DocTableInfo;
+import io.crate.types.DataTypes;
 
 import java.util.Collection;
 import java.util.Locale;
+import java.util.Map;
 
 public final class ConstraintsValidator {
 
-    public static void validate(Object value, Reference targetColumn) {
+    @SuppressWarnings("unchecked")
+    public static void validate(Object value, Reference targetColumn, DocTableInfo tableInfo) {
```

Comment:
Would it make sense to change the dependency from tableInfo to just the notNullColumns?